### DISCUSSION TITLE
Show two boxes side by side

### DISCUSSION THREAD

**Discussion by lvjr** [typst/typst#494] [Q&A]
I tried to implement the following result in @laurmaedje's thesis but failed:

![image](https://user-images.githubusercontent.com/4239949/229258703-d6a3d8ad-4c6f-48ec-9017-424dda40717a.png)

```typst
#table(
columns:(1fr,1fr),
fill: (col, _) => if col == 0 { gray.lighten(70%) },
[```
#let x = 1
#let y = 2
The sum of #x and #y is #{x + y}
```],[
#let x = 1
#let y = 2
The sum of #x and #y is #{x + y}  
]
)
```
![image](https://user-images.githubusercontent.com/4239949/229259150-63c3e592-47b1-4f28-8a4e-6dfbc6d99c63.png)

The missing feature is the round corners, since Typst doesn't support round corners in tables. Any suggestions? Thank you.

**Comment by ararunaufc** [comment]
You could use `rect`, which has customisable rounded corners and can contain arbitrary `content`.

**Reply by lvjr** [reply]
Could you show me the code to make this?

**Reply by ararunaufc** [reply]
Well, I could only set it up in a hacky way until now, and it has at least two problems:

- no color yet,
- doesn't work if you don't set a `height`

```typst
#let divided(left, right) = {
  rect(
    stroke: 1pt,
    radius: 8pt,
    inset: 0pt,
    height: 3cm,
    grid(
      columns: (1fr, auto, 1fr),
      gutter: 2pt,
      pad(rest: 8pt, left),
      line(end: (0%, 100%)),
      pad(rest: 8pt, right)
    )
  )
}

#divided(raw(lorem(10)), lorem(15))
```

![rendered.png](https://user-images.githubusercontent.com/19910857/229326833-d250f486-82d8-4505-a798-021111a5f79b.png)

**Comment by xkevio** [comment]
This will work. Height calculation is needed as otherwise, one column might be smaller than the other one.
A `table`, by default, has an `inset` of `5pt` so we counter that by setting the `outset` of the `box` to `5pt`.

Could also do that the other way around but then it's way tighter.

```
#let divided(left, right) = {
  style(styles => {
    let left_height = measure(left, styles).height
    let right_height = measure(right, styles).height
    let max_height = calc.max(left_height, right_height)
    
    table(
      columns: (1fr, 1fr),
      stroke: none,
      box(left, fill: gray.lighten(90%), width: 100%, height: max_height,
        outset: 5pt, radius: (left: 5pt), stroke: black),
      box(right, width: 100%, height: max_height,
        outset: 5pt, radius: (right: 5pt), stroke: black)
    )
  })
}
```

**Reply by lvjr** [reply]
With `#divided(lorem(20), lorem(25))` I got the following overflowed result:

![image](https://user-images.githubusercontent.com/4239949/229460156-c3220dbf-357c-473d-833f-3d0863aaa8e0.png)

**Reply by xkevio** [reply]
Hm yea I see. Turns out, `measure` currently isn't really capable of considering the layout like that when measuring width or height.
Right now, the easiest way is probably just manually specifying the height of the `box`es -- @laurmaedje did as follows in his thesis:

> "The only way I know of (which I used in my thesis), is to have both contents on both sides, overlay them (e.g. with a grid) and hide one on each side. "
